maildir.go

```diff
@@ -502,6 +500,44 @@ func (d Dir) Clean() error {
 	return nil
 }
 
+// Walk calls fn for every messages in the mailbox's subdir directory.
+//
+// subdir should be "cur", "new" or "tmp".
```

Comment:
I'm not comfortable exposing the subdir directly to the API user. I think only iterating over "cur" is desirable?